crabs/detection_tracking/train_model.py

```diff
@@ -163,10 +169,46 @@ def core_training(self) -> lightning.Trainer:
 
         # Get model
         lightning_model = FasterRCNN(self.config)
-
         # Run training
-        trainer = self.setup_trainer()
-        trainer.fit(lightning_model, data_module)
+        trainer, checkpoint_callback = self.setup_trainer()
+
+        if self.checkpoint_path and os.path.exists(self.checkpoint_path):
+            logging.info(
+                f"Checking contents of checkpoint: {self.checkpoint_path}"
+            )
+            checkpoint = torch.load(self.checkpoint_path)
+
+            if (
+                "optimizer_states" in checkpoint
+                and "lr_schedulers" in checkpoint
+            ):
+                logging.info(
+                    "Checkpoint contains full training state. Resume training"
+                )
+                # Resume training from the checkpoint
+                trainer.fit(
+                    lightning_model,
+                    data_module,
+                    ckpt_path=self.checkpoint_path,
+                )
+            else:
+                logging.info(
+                    "Checkpoint contains only model weights. Load the weight from a trained model"
+                )
+                # Load model weights and start fine-tuning
+                model = FasterRCNN.load_from_checkpoint(self.checkpoint_path)
+                trainer.fit(model, data_module)
+
+        else:
+            trainer.fit(
+                lightning_model,
+                data_module,
+            )
+
+        if checkpoint_callback and checkpoint_callback.last_model_path:
+            logging.info(
+                f"Last checkpoint path: {checkpoint_callback.last_model_path}"
+            )
```

Comment:
I think we can remove this bit and avoid having to return `checkpoint_callback` in `setup_trainer()`.
```suggestion

```